apps/cloudbroker/base/cloudapi__disks/methodclass/cloudapi_disks.py

```diff
@@ -246,3 +252,21 @@ def resize(self, diskId, size, **kwargs):
         if not res:
             raise exceptions.Accepted(False)
         return True
+
+    @authenticator.auth(acl={'cloudspace': set('X')})
+    def restore(self, diskId, reason, **kwargs):
+        disk = self.models.disk.searchOne({'id': diskId})
+        if not disk or disk['status'] == resourcestatus.Disk.DESTROYED:
+            raise exceptions.NotFound("Couldn't find disk with id: %s" % diskId)
+        if disk['status'] != resourcestatus.Disk.TOBEDELETED:
```

Comment:
werent we gonna change this enum name?

Reply:
I changed the behavior now. `DELETED` state is for deleted attached disks and `TOBEDELETED` for unattached disks. I know this is not optimal and i am open to suggestions.